### PR TITLE
feat(logify): Add logify and logifyAll methods

### PR DESCRIPTION
**Pull request by fbeuvens**
Move to es6

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/Sagacify/logger/pull/2" target="_blank">ReviewNinja

File: src/config.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 var env = process.env;
```
Comment (SagaStalin) [inline src/config.js]: MAJOR Replace "var" with "let" or "const" [rule](https://sonar.sagacify.com/coding_rules#rule_key=javascript%3AS3504)

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          reject(err);
+        });
+    });
+  };
+
+  const logifyAll = (obj, promisified = false) => {
+    if (promisified === true) {
+      promisified = 'Async';
+    }
+    for (let key in obj) {
```
Comment (SagaStalin) [inline src/index.js]: MINOR Make "key" "const". [rule](https://sonar.sagacify.com/coding_rules#rule_key=javascript%3AS3353)

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      }
+    }
+    return obj;
+  };
+
+  return Object.assign(logMethods, { logify, logifyAll });
+};
+
+export const create = info => {
+  var loggerRef = mainLogger.child(info);
```
Comment (SagaStalin) [inline src/index.js]: MAJOR Replace "var" with "let" or "const" [rule](https://sonar.sagacify.com/coding_rules#rule_key=javascript%3AS3504)

File: src/package.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const loadMainPackageJSON = (attempts = 1) => {
+  if (attempts > 5) {
+    throw new Error('Can\'t resolve main package.json file');
+  }
+  const mainPath = attempts === 1 ? './' : Array(attempts).join('../');
```
Comment (SagaStalin) [inline src/package.js]: MINOR Convert this usage of the ternary operator to an "if"/"else" structure. [rule](https://sonar.sagacify.com/coding_rules#rule_key=javascript%3AConditionalOperator)

**Comment by SagaStalin** [comment]
SonarQube analysis reported 4 issues
* MAJOR 2 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      }
+    }
+    return obj;
+  };
+
+  return _.assign(logMethods, { logify, logifyAll });
+};
+
+export const create = info => {
+  var loggerRef = mainLogger.child(info);
```
Comment (SagaStalin) [inline src/index.js]: MAJOR Replace "var" with "let" or "const" [rule](https://sonar.sagacify.com/coding_rules#rule_key=javascript%3AS3504)

**Comment by SagaStalin** [comment]
SonarQube analysis reported 2 issues
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: README.md [hunk 6]
```diff
@@ -33,3 +33,13 @@ log.error(event, data, metadata);
...
+It is also possible to logify a function returning a promise:
+```js
+log.logify(func, event);
+```
+
+To replace all functions returning promise of an object:
+```js
+log.logifyAll(obj, promisified);
+```
+Promisified is optional and is used to pass the suffix of methods generated with the Bluebird promisifyAll method.
```
Comment (aborsu) [inline README.md]: An example would be nice here.
I'm not sure what you mean by this.

**Pull request merged**